### PR TITLE
Add common linters to tf2

### PR DESCRIPTION
**Pull request by ahcorde**
This PR adds common linters. It builds on top of this other PR https://github.com/ros2/geometry2/pull/257. 

 - uncrustify 
 - cppcheck
 - cpplint 
 - copyright headers

### CODE REVIEW

**Comment by ahcorde** [comment]
I still have some issues with some copyright headers located in [tf2/LinearMath](https://github.com/ros2/geometry2/blob/ahcorde/docs/improve/tf2/include/tf2/LinearMath/Matrix3x3.h). This copyright header is no defined in the [templates](https://github.com/ament/ament_lint/tree/master/ament_copyright/ament_copyright/template). Any thoughts about how to include this particular copyrights?

**Comment by clalancette** [comment]
> I still have some issues with some copyright headers located in [tf2/LinearMath](https://github.com/ros2/geometry2/blob/ahcorde/docs/improve/tf2/include/tf2/LinearMath/Matrix3x3.h). This copyright header is no defined in the [templates](https://github.com/ament/ament_lint/tree/master/ament_copyright/ament_copyright/template). Any thoughts about how to include this particular copyrights?

Ug.  That looks like some kind of MIT derivative, or something.  We'll have to do some research on it; for now, I'll suggest that we just disable copyright checking.

File: tf2/include/tf2/LinearMath/Matrix3x3.h [hunk 1]
```diff
@@ -23,610 +20,604 @@ subject to the following restrictions:
...
-	void	deSerializeFloat(const struct	Matrix3x3FloatData& dataIn);
-
-        TF2_PUBLIC
-	void	deSerializeDouble(const struct	Matrix3x3DoubleData& dataIn);
-
+  /** @brief No initializaion constructor */
+  TF2_PUBLIC
+  Matrix3x3() {}
+
+  // explicit Matrix3x3(const tf2Scalar *m) { setFromOpenGLSubMatrix(m); }
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Matrix3x3.h]: While we are in here, let's just remove this; it is commented out in both ROS 1 and ROS 2, and thus I think it is safe to say nobody needs it right now.

File: tf2/include/tf2/LinearMath/Matrix3x3.h [hunk 2]
```diff
@@ -23,610 +20,604 @@ subject to the following restrictions:
...
+  /**@brief Constructor from Quaternion */
+  TF2_PUBLIC
+  explicit Matrix3x3(const Quaternion & q) {setRotation(q);}
+  /*
+  template <typename tf2Scalar>
+  Matrix3x3(const tf2Scalar& yaw, const tf2Scalar& pitch, const tf2Scalar& roll)
+  {
+  setEulerYPR(yaw, pitch, roll);
+  }
+  */
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Matrix3x3.h]: Let's remove this as well.

File: tf2/include/tf2/LinearMath/Matrix3x3.h [hunk 3]
```diff
@@ -23,610 +20,604 @@ subject to the following restrictions:
...
+  {
+    return m_el[0].y() * v.x() + m_el[1].y() * v.y() + m_el[2].y() * v.z();
+  }
+  TF2SIMD_FORCE_INLINE tf2Scalar tdotz(const Vector3 & v) const
+  {
+    return m_el[0].z() * v.x() + m_el[1].z() * v.y() + m_el[2].z() * v.z();
+  }
+
+
+  /**@brief diagonalizes this matrix by the Jacobi method.
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Matrix3x3.h]: Nit: shouldn't this be the "Jacobian" method?

File: tf2/include/tf2/LinearMath/QuadWord.h [hunk 4]
```diff
@@ -37,149 +34,144 @@ class QuadWord
...
-	TF2SIMD_FORCE_INLINE	bool	operator!=(const QuadWord& other) const
-	{
-		return !(*this == other);
-	}
-
-  /**@brief Set x,y,z and zero w 
+  TF2SIMD_FORCE_INLINE const tf2Scalar & w() const {return m_floats[3];}
+
+  // TF2SIMD_FORCE_INLINE tf2Scalar&       operator[](int i)       { return (&m_floats[0])[i]; }
+  // TF2SIMD_FORCE_INLINE const tf2Scalar& operator[](int i) const { return (&m_floats[0])[i]; }
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/QuadWord.h]: I think it is safe to remove these as well.

File: tf2/include/tf2/LinearMath/Quaternion.h [hunk 5]
```diff
@@ -25,452 +20,477 @@ subject to the following restrictions:
...
   /**@brief No initialization constructor */
-        TF2_PUBLIC
-	Quaternion() {}
+  TF2_PUBLIC
+  Quaternion() {}
 
-	//		template <typename tf2Scalar>
-	//		explicit Quaternion(const tf2Scalar *v) : Tuple4<tf2Scalar>(v) {}
+  // template <typename tf2Scalar>
+  // explicit Quaternion(const tf2Scalar *v) : Tuple4<tf2Scalar>(v) {}
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Quaternion.h]: Remove this unused code.

File: tf2/include/tf2/LinearMath/Quaternion.h [hunk 6]
```diff
@@ -25,452 +20,477 @@ subject to the following restrictions:
...
-  /**@brief Return the sum of this quaternion and the other 
+  TF2_PUBLIC
+  tf2Scalar angle(const Quaternion & q) const
+  {
+    tf2Scalar s = tf2Sqrt(length2() * q.length2());
+    tf2Assert(s != tf2Scalar(0.0));
+    return tf2Acos(dot(q) / s);
+  }
+  /**@brief Return the angle between this quaternion and the other along the shortest path
+  * @param q The other quaternion */
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Quaternion.h]: Nit: this comment indentation seems somewhat irregular with the rest of the code.

File: tf2/include/tf2/LinearMath/Scalar.h [hunk 7]
```diff
@@ -1,193 +1,177 @@
...
+  #define TF2SIMD_FORCE_INLINE inline
+  #define ATTRIBUTE_ALIGNED16(a) a
+  #define ATTRIBUTE_ALIGNED64(a) a
+  #define ATTRIBUTE_ALIGNED128(a) a
+#else
+// #define TF2_HAS_ALIGNED_ALLOCATOR
+  #pragma warning(disable : 4324)  // disable padding warning
+// #pragma warning(disable:4530)  // Disable the exception disable but used in MSCV Stl warning.
+// #pragma warning(disable:4996)  // Turn off warnings about deprecated C routines
+// #pragma warning(disable:4786)  // Disable the "debug name too long" warning
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Scalar.h]: Let's just remove these, they clearly aren't needed.

File: tf2/include/tf2/LinearMath/Scalar.h [hunk 8]
```diff
@@ -1,193 +1,177 @@
...
-
-
 #ifdef _WIN32
+  #if defined(__MINGW32__) || defined(__CYGWIN__) || (defined (_MSC_VER) && _MSC_VER < 1300)
+  #define TF2SIMD_FORCE_INLINE inline
+  #define ATTRIBUTE_ALIGNED16(a) a
+  #define ATTRIBUTE_ALIGNED64(a) a
+  #define ATTRIBUTE_ALIGNED128(a) a
+#else
+// #define TF2_HAS_ALIGNED_ALLOCATOR
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Scalar.h]: Let's just remove this for now.

File: tf2/include/tf2/LinearMath/Transform.h [hunk 9]
```diff
@@ -1,321 +1,297 @@
...
+  {
+    return getRotation() * q;
+  }
 
   /**@brief Return the basis matrix for the rotation */
-	TF2SIMD_FORCE_INLINE Matrix3x3&       getBasis()          { return m_basis; }
+  TF2SIMD_FORCE_INLINE Matrix3x3 & getBasis() {return m_basis;}
   /**@brief Return the basis matrix for the rotation */
-	TF2SIMD_FORCE_INLINE const Matrix3x3& getBasis()    const { return m_basis; }
+  TF2SIMD_FORCE_INLINE const Matrix3x3 & getBasis()    const {return m_basis;}
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Transform.h]: We should use only one space before the `const`.

File: tf2/include/tf2/LinearMath/Transform.h [hunk 10]
```diff
@@ -1,321 +1,297 @@
...
 
   /**@brief Return the origin vector translation */
-	TF2SIMD_FORCE_INLINE Vector3&         getOrigin()         { return m_origin; }
+  TF2SIMD_FORCE_INLINE Vector3 & getOrigin() {return m_origin;}
   /**@brief Return the origin vector translation */
-	TF2SIMD_FORCE_INLINE const Vector3&   getOrigin()   const { return m_origin; }
+  TF2SIMD_FORCE_INLINE const Vector3 & getOrigin()   const {return m_origin;}
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Transform.h]: We should use only one space before the `const`.

File: tf2/include/tf2/LinearMath/Transform.h [hunk 11]
```diff
@@ -1,321 +1,297 @@
...
+  TF2_PUBLIC
+  void    deSerialize(const struct        TransformData & dataIn);
 
-        TF2_PUBLIC
-	void	deSerializeFloat(const struct	TransformFloatData& dataIn);
+  TF2_PUBLIC
+  void    deSerializeDouble(const struct  TransformDoubleData & dataIn);
 
+  TF2_PUBLIC
+  void    deSerializeFloat(const struct   TransformFloatData & dataIn);
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Transform.h]: Can we just make the spacing consistent with the rest of the code?  That is, one space after the return type, and one space separating the argument qualifiers?

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 12]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
-		{
-			tf2SetMax(m_floats[0], other.m_floats[0]);
-			tf2SetMax(m_floats[1], other.m_floats[1]);
-			tf2SetMax(m_floats[2], other.m_floats[2]);
-			tf2SetMax(m_floats[3], other.w());
-		}
+  TF2SIMD_FORCE_INLINE const tf2Scalar & w() const {return m_floats[3];}
+
+  // TF2SIMD_FORCE_INLINE tf2Scalar&       operator[](int i)       { return (&m_floats[0])[i];}
+  // TF2SIMD_FORCE_INLINE const tf2Scalar& operator[](int i) const { return (&m_floats[0])[i]; }
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Let's just remove this unused code.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 13]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+  }
+
+  TF2SIMD_FORCE_INLINE bool operator!=(const Vector3 & other) const
+  {
+    return !(*this == other);
+  }
+
+  /**@brief Set each element to the max of the current values and the values of another Vector3
+* @param other The other Vector3 to compare with
+*/
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird indentation on the comment.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 14]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
-		TF2SIMD_FORCE_INLINE	void	deSerialize(const struct	Vector3Data& dataIn);
-
-		TF2SIMD_FORCE_INLINE	void	serializeFloat(struct	Vector3FloatData& dataOut) const;
-
-		TF2SIMD_FORCE_INLINE	void	deSerializeFloat(const struct	Vector3FloatData& dataIn);
-
-		TF2SIMD_FORCE_INLINE	void	serializeDouble(struct	Vector3DoubleData& dataOut) const;
-
-		TF2SIMD_FORCE_INLINE	void	deSerializeDouble(const struct	Vector3DoubleData& dataIn);
+  TF2SIMD_FORCE_INLINE void       setMin(const Vector3 & other)
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing between return type and method name.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 15]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+  {
+    tf2SetMin(m_floats[0], other.m_floats[0]);
+    tf2SetMin(m_floats[1], other.m_floats[1]);
+    tf2SetMin(m_floats[2], other.m_floats[2]);
+    tf2SetMin(m_floats[3], other.w());
+  }
+
+  TF2SIMD_FORCE_INLINE void       setValue(
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing between return type and method name.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 16]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+    const tf2Scalar & z)
+  {
+    m_floats[0] = x;
+    m_floats[1] = y;
+    m_floats[2] = z;
+    m_floats[3] = tf2Scalar(0.);
+  }
+
+  TF2_PUBLIC
+  void    getSkewSymmetricMatrix(Vector3 * v0, Vector3 * v1, Vector3 * v2) const
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing between return type and method name.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 17]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+  {
+    v0->setValue(0., -z(), y());
+    v1->setValue(z(), 0., -x());
+    v2->setValue(-y(), x(), 0.);
+  }
+
+  TF2_PUBLIC
+  void    setZero()
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing between return type and method name.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 18]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+
+  TF2SIMD_FORCE_INLINE void deSerialize(const struct        Vector3Data & dataIn);
+
+  TF2SIMD_FORCE_INLINE void serializeFloat(struct   Vector3FloatData & dataOut) const;
+
+  TF2SIMD_FORCE_INLINE void deSerializeFloat(const struct   Vector3FloatData & dataIn);
+
+  TF2SIMD_FORCE_INLINE void serializeDouble(struct  Vector3DoubleData & dataOut) const;
+
+  TF2SIMD_FORCE_INLINE void deSerializeDouble(const struct  Vector3DoubleData & dataIn);
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing between type and qualifiers.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 19]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
 
-TF2SIMD_FORCE_INLINE Vector3 Vector3::rotate( const Vector3& wAxis, const tf2Scalar angle ) const
+TF2SIMD_FORCE_INLINE Vector3 Vector3::rotate(const Vector3 & wAxis, const tf2Scalar angle) const
 {
-	// wAxis must be a unit lenght vector
-
-	Vector3 o = wAxis * wAxis.dot( *this );
-	Vector3 x = *this - o;
-	Vector3 y;
+  // wAxis must be a unit lenght vector
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]:
```suggestion
  // wAxis must be a unit length vector
```

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 20]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+  {
+    return tf2Vector4(
+      tf2Fabs(m_floats[0]),
+      tf2Fabs(m_floats[1]),
+      tf2Fabs(m_floats[2]),
+      tf2Fabs(m_floats[3]));
+  }
+
+  TF2_PUBLIC
+  tf2Scalar       getW() const {return m_floats[3];}
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 21]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
-		TF2SIMD_FORCE_INLINE void	setValue(const tf2Scalar& x, const tf2Scalar& y, const tf2Scalar& z,const tf2Scalar& w)
-		{
-			m_floats[0]=x;
-			m_floats[1]=y;
-			m_floats[2]=z;
-			m_floats[3]=w;
-		}
-
-
+  TF2SIMD_FORCE_INLINE void       setValue(
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 22]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+    m_floats[3] = w;
+  }
 };
 
-
-///tf2SwapVector3Endian swaps vector endianness, useful for network and cross-platform serialization
-TF2SIMD_FORCE_INLINE void	tf2SwapScalarEndian(const tf2Scalar& sourceVal, tf2Scalar& destVal)
+// tf2SwapVector3Endian swaps vector endianness, useful for network and
+// cross-platform serialization
+TF2SIMD_FORCE_INLINE void       tf2SwapScalarEndian(
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 23]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+  dest[4] = src[3];
+  dest[5] = src[2];
+  dest[6] = src[1];
+  dest[7] = src[0];
 }
-///tf2SwapVector3Endian swaps vector endianness, useful for network and cross-platform serialization
-TF2SIMD_FORCE_INLINE void	tf2SwapVector3Endian(const Vector3& sourceVec, Vector3& destVec)
+// tf2SwapVector3Endian swaps vector endianness, useful for network and
+// cross-platform serialization
+TF2SIMD_FORCE_INLINE void       tf2SwapVector3Endian(const Vector3 & sourceVec, Vector3 & destVec)
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 24]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
+  for (int i = 0; i < 4; i++) {
+    tf2SwapScalarEndian(sourceVec[i], destVec[i]);
+  }
 }
 
-///tf2UnSwapVector3Endian swaps vector endianness, useful for network and cross-platform serialization
-TF2SIMD_FORCE_INLINE void	tf2UnSwapVector3Endian(Vector3& vector)
+// tf2UnSwapVector3Endian swaps vector endianness, useful for network and
+// cross-platform serialization
+TF2SIMD_FORCE_INLINE void       tf2UnSwapVector3Endian(Vector3 & vector)
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 25]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
-struct	Vector3DoubleData
+struct  Vector3DoubleData
 {
-	double	m_floats[4];
-
+  double m_floats[4];
 };
 
-TF2SIMD_FORCE_INLINE	void	Vector3::serializeFloat(struct	Vector3FloatData& dataOut) const
+TF2SIMD_FORCE_INLINE void Vector3::serializeFloat(struct  Vector3FloatData & dataOut) const
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]:
```suggestion
TF2SIMD_FORCE_INLINE void Vector3::serializeFloat(struct Vector3FloatData & dataOut) const
```

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 26]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
-	for (int i=0;i<4;i++)
-		dataOut.m_floats[i] = float(m_floats[i]);
+  // could also do a memcpy, check if it is worth it
+  for (int i = 0; i < 4; i++) {
+    dataOut.m_floats[i] = static_cast<float>(m_floats[i]);
+  }
 }
 
-TF2SIMD_FORCE_INLINE void	Vector3::deSerializeFloat(const struct	Vector3FloatData& dataIn)
+TF2SIMD_FORCE_INLINE void Vector3::deSerializeFloat(const struct  Vector3FloatData & dataIn)
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]:
```suggestion
TF2SIMD_FORCE_INLINE void Vector3::deSerializeFloat(const struct Vector3FloatData & dataIn)
```

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 27]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
-	for (int i=0;i<4;i++)
-		m_floats[i] = tf2Scalar(dataIn.m_floats[i]);
+  for (int i = 0; i < 4; i++) {
+    m_floats[i] = tf2Scalar(dataIn.m_floats[i]);
+  }
 }
 
-
-TF2SIMD_FORCE_INLINE	void	Vector3::serialize(struct	Vector3Data& dataOut) const
+TF2SIMD_FORCE_INLINE void Vector3::serialize(struct       Vector3Data & dataOut) const
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing.

File: tf2/include/tf2/LinearMath/Vector3.h [hunk 28]
```diff
@@ -28,708 +23,676 @@ namespace tf2
...
-	for (int i=0;i<4;i++)
-		dataOut.m_floats[i] = m_floats[i];
+  // could also do a memcpy, check if it is worth it
+  for (int i = 0; i < 4; i++) {
+    dataOut.m_floats[i] = m_floats[i];
+  }
 }
 
-TF2SIMD_FORCE_INLINE void	Vector3::deSerialize(const struct	Vector3Data& dataIn)
+TF2SIMD_FORCE_INLINE void Vector3::deSerialize(const struct       Vector3Data & dataIn)
```
Comment (clalancette) [inline tf2/include/tf2/LinearMath/Vector3.h]: Weird spacing.

File: tf2/include/tf2/buffer_core.h [hunk 29]
```diff
@@ -1,56 +1,54 @@
...
-//#include "geometry_msgs/TwistStamped.h"
-#include "geometry_msgs/msg/transform_stamped.hpp"
-
 #include <map>
 #include <unordered_map>
 #include <mutex>
 #include <functional>
 #include <memory>
+#include <utility>
+#include <vector>
```
Comment (clalancette) [inline tf2/include/tf2/buffer_core.h]: Nit: alphabetize the list.

File: tf2/include/tf2/time.h [hunk 30]
```diff
@@ -1,97 +1,97 @@
...
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef TF2__TIME_H_
+#define TF2__TIME_H_
+
+#include <stdio.h>
```
Comment (clalancette) [inline tf2/include/tf2/time.h]: I think we can probably remove this completely; we aren't using it in this file, I don't think.

File: tf2/include/tf2/time.h [hunk 31]
```diff
@@ -1,97 +1,97 @@
...
 
 #include <chrono>
 #include <cmath>
-#include <stdio.h>
 #include <string>
 #include <thread>
 #include <iostream>
```
Comment (clalancette) [inline tf2/include/tf2/time.h]: I don't think we need either of these in this header file.

File: tf2/include/tf2/time_cache.h [hunk 32]
```diff
@@ -1,120 +1,125 @@
...
 /** \author Tully Foote */
 
-#ifndef TF2_TIME_CACHE_H
-#define TF2_TIME_CACHE_H
-
-#include "transform_storage.h"
+#ifndef TF2__TIME_CACHE_H_
+#define TF2__TIME_CACHE_H_
 
 #include <memory>
```
Comment (clalancette) [inline tf2/include/tf2/time_cache.h:34]: Add `#include <chrono>` to this list.

File: tf2/include/tf2/transform_storage.h [hunk 33]
```diff
@@ -52,17 +46,18 @@ class TransformStorage
...
+  TransformStorage(const TransformStorage & rhs)
   {
     *this = rhs;
   }
 
   TF2_PUBLIC
-  TransformStorage& operator=(const TransformStorage& rhs)
+  TransformStorage & operator=(const TransformStorage & rhs)
   {
 #if 01
```
Comment (clalancette) [inline tf2/include/tf2/transform_storage.h:62]: This looks like a bug.  If you don't want to deal with it in this PR, please open an issue to follow-up on it.

File: tf2/src/buffer_core.cpp [hunk 34]
```diff
@@ -44,11 +50,18 @@ namespace tf2
 static double QUATERNION_NORMALIZATION_TOLERANCE = 10e-3;
```
Comment (clalancette) [inline tf2/src/buffer_core.cpp]: `constexpr`, maybe?

File: tf2/src/buffer_core.cpp [hunk 35]
```diff
@@ -163,173 +184,187 @@ CompactFrameID BufferCore::validateFrameId(
...
-                                  const std::string child_frame_id, const TimePoint stamp,
-                                  const std::string& authority, bool is_static)
+bool BufferCore::setTransformImpl(
+  const tf2::Transform & transform_in, const std::string frame_id,
+  const std::string child_frame_id, const TimePoint stamp,
+  const std::string & authority, bool is_static)
 {
-
-  /////BACKEARDS COMPATABILITY 
+  // BACKEARDS COMPATABILITY
```
Comment (clalancette) [inline tf2/src/buffer_core.cpp]:
```suggestion
  // BACKWARDS COMPATABILITY
```

**Review by clalancette — COMMENTED** [review]
Overall, this looks pretty good.  A lot of little nits inline.  There are also some bigger things listed below.  But since I don't want this PR to get much bigger, please open issues for these so we can follow-up later:

1.  Rename all `.h` -> `.hpp` files.  This will have to also have a deprecation warning, and should be done post-Foxy.
1.  Go through the comments (particularly API comments) and fix the ones that say `todo`.
1.  Go through the comments and use a common Doxygen style (some of the comments use `\` and some use `@`).

**Comment by ahcorde** [comment]
Thanks for the review! it's quite long the PR.

The issue was created https://github.com/ros2/geometry2/issues/259

File: tf2/include/tf2/LinearMath/Matrix3x3.h (hunk 3 above)
Comment (ahcorde) [reply tf2/include/tf2/LinearMath/Matrix3x3.h]: https://en.wikipedia.org/wiki/Jacobi_method

**Comment by ahcorde** [comment]
For now I will not remove  because it's part of the public API. Maybe in the post-foxy release review.

File: tf2/include/tf2/LinearMath/Matrix3x3.h (hunk 3 above)
Comment (clalancette) [reply tf2/include/tf2/LinearMath/Matrix3x3.h]: Ah, never mind then :).

File: tf2/include/tf2/time.h (hunk 30 above)
Comment (clalancette) [reply tf2/include/tf2/time.h]: This comment about removing stdio.h is still open.

File: tf2/include/tf2/time.h (hunk 31 above)
Comment (clalancette) [reply tf2/include/tf2/time.h]: This comment about removing `thread` and `iostream` is still open.

**Review by clalancette — COMMENTED** [review]
Two more minor things to fix, then I think we're good to go.

One question: in #259, what do you mean by "remove time.h"?  I think we added that relatively recently so that people could just get the time conversions, but maybe I'm misremembering.  Can you elaborate more on why?

**Comment by ahcorde** [comment]
I understood the comment wrong. I read that the "file" was not used,  not just the `#include <stdio.h>`. Sorry about that, I updated the issue

**Review by clalancette — APPROVED** [review]
Looks good with green CI.

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10559/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6012/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8599/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10441/)

**Comment by tfoote** [comment]
I'd really like to not apply the linters to LinearMath files. It's a fork of Bullets LinearMath implementation and we periodically sync between them. Can we suppress the linting in that subdirectory?

**Comment by ahcorde** [comment]
For now we can only suppress the linting in uncrustify, but with this PR https://github.com/ament/ament_lint/pull/234 we can suppress them also in cppcheck and cpplint

**Review by tfoote — APPROVED** [review]
Thanks this looks much better with excluding the LinearMath content.

**Comment by clalancette** [comment]
So, while I think we're good to go here, I would advocate for not merging this PR until after we branch off for foxy.  The risk of regression is fairly low, but it's not zero and I think we should let foxy calm down a bit.  @ahcorde @tfoote what do you think?

**Comment by dirk-thomas** [comment]
Since Foxy will be supported for 3+ years I would highly suggest to not have such a big diff between the Foxy and G-turtle branch.

**Comment by clalancette** [comment]
> Since Foxy will be supported for 3+ years I would highly suggest to not have such a big diff between the Foxy and G-turtle branch.

Why?

**Comment by dirk-thomas** [comment]
Because of the extra effort to backport fixes.

**Comment by ahcorde** [comment]
This package is one of the candidates for Quality Level 1 or 2. This means that [linters is a must](https://github.com/ros-infrastructure/rep/pull/218/files#diff-f40b01badf3f268cbb5d72ffcc9613f5R617). what's the best way to proceed?

**Comment by clalancette** [comment]
My opinion hasn't changed; if we keep shoveling stuff into Foxy, then it will never get done.

That being said, if others think we should proceed, then I won't stand in the way.

**Comment by tfoote** [comment]
Clearly it's a risk, but I think continuity with G-Turtle and forward  will be more valuable than backwards compatibility with Noetic so the maintenance hopefully will be reduced. We're already pretty diverged from the ROS 1 branch so patches won't just apply across. 

Clearly any big change is a risk but this has reasonable coverage so I think it's relatively low especially with it theoretically being just reformatting.

@ros-pull-request-builder retest this please

**Comment by ahcorde** [comment]
@tfoote the pull-request-builder failed this other PR should resolve the issue https://github.com/ros2/geometry2/pull/267

**Comment by ahcorde** [comment]
@ros-pull-request-builder retest this please

**Comment by ahcorde** [comment]
Testing changes on ament_lint https://github.com/ament/ament_lint/pull/238. Running CI up-to tf2

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10940/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6286/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8909/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10843/)

**Comment by clalancette** [comment]
I've now rebased and force-pushed this on the latest.  It still needs the changes from ament/ament_lint#238 to go in first, though.

**Comment by clalancette** [comment]
I've rebased this onto the latest, and also replaced the actual linting checks with a TODO.

I'd like to go ahead and merge this, since this is going to affect any incoming PRs that touch the `tf2` directory.  We won't have the actual checks in place until we resolve https://github.com/ament/ament_lint/pull/238 , but it should be a *much* smaller patch once we do that.

With all of that said, then, here is another CI up to tf2.  If this succeeds, I'll merge:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11788/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6871/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9626/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11732/)

**Pull request merged**